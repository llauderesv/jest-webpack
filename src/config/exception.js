// Extending Error classes.
// Handling validation exception in api request

// Super keywords used to call the base class properties
// w/out calling super the this property of a class is undefined.
export class ValidationException extends Error {
  constructor(message) {
    super(message);
    this.name = 'ValidationException';
  }
}
