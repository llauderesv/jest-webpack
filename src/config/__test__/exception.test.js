import { ValidationException } from '../exception';

describe('Exception Suite', () => {
  it('Should throw validation exception', () => {
    try {
      throw new ValidationException('Invalid parameter passed.');
    } catch (error) {
      expect(error.message).toMatchSnapshot();
      expect(error.name).toMatchSnapshot();
    }
  });
});
