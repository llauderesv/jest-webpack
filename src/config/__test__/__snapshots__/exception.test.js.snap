// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exception Suite Should throw validation exception 1`] = `"Invalid parameter passed."`;

exports[`Exception Suite Should throw validation exception 2`] = `"ValidationException"`;
